Написать документацию и подготовить проект к публикации в реестре пакетов

postcretinism opened the pull request.

postcretinism committed:
подключили docfx
настроили генерацию сайта

postcretinism committed:
сделали конвертеры internal
прописали комментарии для енама облачности
изменили ридми и сделали в нем новый туду
добавили новые настройки визуализации в docfx.json (убрали кнопку редактирования, начали сортировать енамы по числу, вложенное отображение пространства имен, убрали визуализацию атрибутов сериализации)

postcretinism committed:
дописали документацию для енама Condition

postcretinism committed:
расписал документацию к енамам
расписал описание класса YandexWeatherClient и его конструкторов

postcretinism committed:
описали документацию к исключению и клиенту
переименовали исключение

postcretinism committed:
описали документацию ForecastResponse, IWeatherData, LocationInfo, TimeZoneInfo, WeatherFactData
переименовали Info в Location

postcretinism committed:
описали документацию к DayPartsWeatherData и ForecastData

postcretinism committed:
Описали документацию к модели:
DayPartShortWeatherData, DayPartWeatherData, DayPartsWeatherData, HourWeatherData
Удалили README
отформатировали проект
Написали начало базовой документации

postcretinism committed:
Дописали документацию
подписали тип YandexWeatherClient для логеров

postcretinism committed:
Сложили документацию в отдельную папку

dima117 commented:
- логотип и фавиконка
- лайаут для индекса + разбить на части
  - убрать лайаут из index.md + добавить index.md в оглавление
- убрать ссылку на github и поиск по оглавлению
  - `"_disableTocFilter": true,`
  - `"_disableContribution": true,`
- починить предупреждения
  - добавить `### YamlMime:ManagedReference` в filterConfig
  - поменять регистр в названии исключения
- у `DayPartWeatherData` нет описания
- добавить бейджи
  - [![NuGet Pre Release](https://img.shields.io/nuget/vpre/ThinkingHome.Migrator.Framework.svg)](https://www.nuget.org/packages?q=thinkinghome.migrator)
  - [![license](https://img.shields.io/github/license/thinking-home/migrator.svg)](https://github.com/thinking-home/migrator/blob/master/LICENSE) (и добавить лицензию)
  - убрать из репозитория `_site` и `_api`, сделать [публикацию на github pages](https://dotnet.github.io/docfx/index.html#publish-to-github-pages)

postcretinism committed:
добавили и переименовали разделы в docs и файлы к ним для правильного отображение частей документации
распределили контент из index.md по разделам

postcretinism committed:
добавили описание DayPartWeatherData
исправили некорректный регистр букв в названии исключения
убрали поиск по оглавлению и сслыку на гитхаб
починили предупреждениея при сборке сайта

postcretinism committed:
добавили бейджи про лицензию и версию пакета
добавили лицензию

postcretinism committed:
Добавили лого и фавиконку
поменяли заголовок

postcretinism committed:
закоммитили изображения
переименовали api в _api

postcretinism committed:
доавили конфиг CI

dima117 commented on ThinkingHome.Weather.Api/ThinkingHome.Weather.Api.csproj:
```diff
@@ -7,11 +7,11 @@
     <ItemGroup>
-      <None Remove="TODO.md" />
+        <None Remove="TODO.md"/>
     </ItemGroup>
```
предлагаю убрать

postcretinism committed:
правки по ревью
